### ISSUE TITLE
Add documentation to the code

### ISSUE THREAD

**Issue by courtneypacheco** [kubernetes-sigs/node-feature-discovery-operator#64]
**What would you like to be added**:
Documentation for the code

**Why is this needed**:
I would like to contribute to this codebase, but I am not familiar with any of the NFD code right now. So, I'm going to have to ramp up and learn how everything works, but I figure that if I'm going to learn the codebase, I might as well update the documentation by adding comments to functions, etc in the process. Then new users who want to contribute to this codebase will be able to ramp up quickly, which will hopefully encourage more people to contribute to this repo.

Thoughts?

**Comment by marquiz** [comment]
Welcome @courtneypacheco! I think improving/adding docstrings and adding comments where you find the code hard to follow is very much appreciated

**Comment by marquiz** [comment]
Fixed by #87, I'd say
/close
